woocommerce/woocommerce
# Deprecate `woocommerce_admin_reports` filter for legacy reports

_requires https://github.com/woocommerce/woocommerce/pull/51557_
### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.



### Changes proposed in this Pull Request:





<details>
<summary>Previous impementation</summary>

Deprecate `woocommerce_admin_reports` filter for legacy reports use
Stay silent for use for `plugins/woocommerce/src/Admin/API/Reports/Controller.php:111` or any use that does not touch legacy `$reports` structure. See https://github.com/woocommerce/woocommerce/issues/36737

This is a somewhat imperfect solution, as we cannot identify a use of `woocommerce_admin_reports` filter that expects legacy reports but does read-only operations.

Alternatively, we could consider deprecating `woocommerce_admin_reports` for all usage and introduce a new filter name for Analytics only. But IMO, that would introduce more confusion and inconvenience for 3PDs and merchants.

I also decided not to throw a deprecation warning from the Analytics controller in case there are some legacy "reports" changes in the filter result, as those were already ignored there.

</details>

Add `Deprecated` tag to `woocommerce_admin_reports`'s PHPDoc. Do not throw deprecation warning, as we cannot precisely and efficiently distinguish between Reports and Analytics usage.


I made a separate PR that prevents errors on legacy reports pages https://github.com/woocommerce/woocommerce/pull/51592. So we could merge it sooner without waiting for the deprecation feature branch. I redo it here so we could test that no deprecation will be shown or Analytics-only use of the `woocommerce_admin_reports` filter.



### How to test the changes in this Pull Request:



Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

1. Read the PHPDoc of `woocommerce_admin_reports` filter.

<details>
<summary>Previous implementation</summary>

1. Checkout this branch and enable `WP_DEBUG`
2. Add a snippet that's intended to use the Analytics filter:
```php
add_filter( 'woocommerce_admin_reports', function( $reports ) {
    $reports[] = array(
        'slug' => 'aw-workflows',
        'description' => 'AutomateWoo Workflows',
        'url' => '/analytics/aw-workflows',
        'path' => '/automatewoo/v1/workflows',
    );
    return $reports;
} );
```
4. Go to the legacy reports page `/wp-admin/admin.php?page=wc-reports&tab=orders`. No warning should be shown.
5. Add a snippet that Changes something in the report's structure
   - Adds another report
```php
add_filter( 'woocommerce_admin_reports', function( $reports ) {
    $reports['automatewoo'] = array(
        'title'   => 'AutomateWoo',
        'reports' => array(
            'runs_by_date' => array(
                'title'       => 'Workflow Runs',
                'description' => 'foo',
                'function'    => 'automatewoo_reports_get_runs_by_date',
            ),
            'conversions' => array(
                'title'       => 'Conversions',
                'description' => 'bar',
                'hide_title'  => true,
            ),
        ),
    );
    return $reports;
} );
```
Check that warning is shown at `/wp-admin/admin.php?page=wc-reports&tab=orders`
- Removes one
```php
add_filter( 'woocommerce_admin_reports', function( $reports ) {
    unset( $reports['stock'] );
    return $reports;
} );
```
Check that warning is shown at `/wp-admin/admin.php?page=wc-reports&tab=orders`
- Makes some changes to the existing ones
```php
add_filter( 'woocommerce_admin_reports', function( $reports ) {
    $reports['stock']['reports']['sales_by_category']['foo'] = 'bar';
    return $reports;
} );
```
Check that warning is shown at `/wp-admin/admin.php?page=wc-reports&tab=orders`
```php
add_filter( 'woocommerce_admin_reports', function( $reports ) {
    unset($reports['stock']['reports']['low_in_stock']['hide_title']);
} );
```
Check that warning is shown at `/wp-admin/admin.php?page=wc-reports&tab=orders`
```php
add_filter( 'woocommerce_admin_reports', function( $reports ) {
    $reports['stock']['reports']['low_in_stock'][] = 'bar';
} );
```
Check that warning is shown at `/wp-admin/admin.php?page=wc-reports&tab=orders`

</details>

		



### Changelog entry

The changelog for this PR is provided in https://github.com/woocommerce/woocommerce/pull/51557







<details>

<summary>Changelog Entry Details</summary>

#### Significance




#### Type




#### Message 

</details>

<details>

<summary>Changelog Entry Comment</summary>

#### Comment 

</details>
